File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/blank_lines/JavaBlankLinesSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.blank_lines;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaBlankLinesSettings.class)
public interface JavaBlankLinesSettings {

  Optional<JavaKeepMaximumBlankLinesSettings> keepMaximumBlankLines();

  Optional<JavaMinimumBlankLinesSettings> minimumBlankLines();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/blank_lines/JavaBlankLinesSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.blank_lines

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaBlankLinesSettings(

    @JsonProperty("Keep Maximum Blank Lines")
    val keepMaximumBlankLines: JavaKeepMaximumBlankLinesSettings? = null,

    @JsonProperty("Minimum Blank Lines")
    val minimumBlankLines: JavaMinimumBlankLinesSettings? = null
)
